File: go/bfs/min_path_test.go
package bfs

import (
	"github.com/stretchr/testify/assert"
	"reflect"
	"testing"
)

func TestMinPath_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.minPath
		actual := minPath(td.graph, td.start, td.end)

		if !reflect.DeepEqual(expected, actual) {
			assert.FailNowf(t, "",
				"minPath(%v, %v, %v):\nwant %v\n got %v",
				td.graph, td.start, td.end, expected, actual,
			)
		}
	}
}

File: python/bfs/min_path_test.py
from min_path import min_path
from random_test_data import RandomTestData


def test_min_path():
	for _ in range(10_000):
		td = RandomTestData()

		want = td.min_path
		got = min_path(td.graph, td.start, td.end)

		if got != want:
			assert False, f'minPath({td.graph}, {td.start}, {td.end}):\nwant {want}\n got {got}'
